teek/_widgets/treeview.py

```diff
@@ -0,0 +1,545 @@
+        return isinstance(other, TreeviewRow) and self._name == other._name
+
+    def _config_handler(self, rettype, option=None, *args):
+        self._check_added()
+
+        if option is not None:
+            args = [option] + list(args)
+
+        return self._treeview._call(
+            rettype, self._treeview, 'item', self, *args
+        )
+
+    def _check_added(self):
+        """
+        Check if row is added to treeview.
+        """
+        if self._treeview is None:
+            raise RuntimeError(
+                "this row hasn't been added to a treeview yet")
+
+    def _assign(self, treeview, root=None, set_config=True):
+        """
+        Internal method for assigning a treeview and a root node to current
+        row.
+        The `set_config` argument specifies if init options should be set.
+        """
+        self._treeview = treeview
+        self._root = root
+
+        self.subrows._treeview = treeview
+
+        if treeview is not None and set_config:
+            for name, value in self._creation_opts.items():
+                self.config[name] = value
+
+    @property
+    def selected(self):
+        """
+        Get current selection state.
+        """
+        self._check_added()
+        return self._name in self._treeview._call([str], self._treeview,
+                                                  'selection')
+
+    @make_thread_safe
+    def select(self):
+        """
+        Add row to current selection.
+        """
+        self._check_added()
+        self._treeview._call(None, self._treeview, 'selection', 'add', self)
+
+    @make_thread_safe
+    def deselect(self):
+        """
+        Remove row from current selection.
+        """
+        self._check_added()
+        self._treeview._call(None, self._treeview, 'selection', 'remove', self)
+
+    @make_thread_safe
+    def move(self, to_pos):
+        """
+        Move row to other position.
+        """
+        self._treeview._call(
+            None, self._treeview, 'move', self, self._root, to_pos
+        )
+
+    def to_tcl(self):
+        # Represented by name
+        return self._name
+
+
+class TreeviewColumnList(MutableSequence):
+    """
+    List containing all columns of a treeview.
+
+    Note, that instances of :class:`TreeviewColumnList` must not be created
+    manually. It is accessible through the `.columns` attribute.
+    """
+    def __init__(self, treeview, init_columns=[]):
+        super().__init__()
+
+        # Default column 0
+        first_column = TreeviewColumn('#0')
+        first_column._assign(treeview)
+
+        self._data = [first_column]
+        self._treeview = treeview
+
+        for column in init_columns:
+            self.append(column)
+
+    def __repr__(self):
+        return '%s(columns=%d)' % (
+            self.__class__.__name__,
+            len(self)
+        )
+
+    @make_thread_safe
+    def __setitem__(self, index, column):
+        if index == 0 or index <= -len(self):
```

Comment:
This check will prevent setting the last column with `columnlist[-1] = something`. Better to use `==` instead of `<=`. There's also a similar thing in `insert`.